be/src/util/arrow/starrocks_column_to_arrow.cpp

```diff
@@ -0,0 +1,309 @@
+        STARROCKS_TO_ARROW_CONV_ENTRY_R(ArrowTypeId::DECIMAL, TYPE_DECIMALV2),
+        STARROCKS_TO_ARROW_CONV_ENTRY_R(ArrowTypeId::DECIMAL, TYPE_DECIMAL32, TYPE_DECIMAL64, TYPE_DECIMAL128),
+        STARROCKS_TO_ARROW_CONV_ENTRY_R(ArrowTypeId::STRING, TYPE_VARCHAR, TYPE_CHAR, TYPE_HLL),
+        STARROCKS_TO_ARROW_CONV_ENTRY_R(ArrowTypeId::STRING, TYPE_LARGEINT, TYPE_DATE, TYPE_DATETIME),
+};
+static inline StarRocksToArrowConvertFunc resolve_convert_func(PrimitiveType pt, ArrowTypeId at, bool is_nullable) {
+    const auto func_id = starrocks_to_arrow_convert_idx(pt, at, is_nullable);
+    const auto end = global_starrocks_to_arrow_conv_table.end();
+    auto it = global_starrocks_to_arrow_conv_table.find(func_id);
+    return it != end ? it->second : nullptr;
+}
+
+class ColumnToArrowArrayConverter : public arrow::TypeVisitor {
+public:
+    ColumnToArrowArrayConverter(const ColumnPtr& column, arrow::MemoryPool* pool, PrimitiveType pt,
+                                std::shared_ptr<arrow::Array>& array)
+            : _column(column), _pool(pool), _pt(pt), _array(array) {}
+
+    using arrow::TypeVisitor::Visit;
+
+#define DEF_VISIT_METHOD(Type)                                                       \
+    arrow::Status Visit(const arrow::Type& type) override {                          \
+        auto func = resolve_convert_func(_pt, type.type_id, _column->is_nullable()); \
+        if (func == nullptr) {                                                       \
+            return arrow::Status::NotImplemented("");                                \
+        }                                                                            \
+        return func(_column, _pool, _array);                                         \
+    }
+
+    DEF_VISIT_METHOD(Decimal128Type);
+    DEF_VISIT_METHOD(DoubleType);
+    DEF_VISIT_METHOD(FloatType);
+    DEF_VISIT_METHOD(Int8Type);
+    DEF_VISIT_METHOD(Int16Type);
+    DEF_VISIT_METHOD(Int32Type);
+    DEF_VISIT_METHOD(Int64Type);
+    DEF_VISIT_METHOD(StringType);
+
+#undef DEF_VISIT_METHOD
+
+private:
+    const ColumnPtr& _column;
+    arrow::MemoryPool* _pool;
+    PrimitiveType _pt;
+    std::shared_ptr<arrow::Array>& _array;
+};
+
+Status vectorized_convert_to_arrow_batch(Chunk* chunk, const std::vector<const TypeDescriptor*>& slot_types,
+                                         const std::vector<SlotId>& slot_ids,
+                                         const std::shared_ptr<arrow::Schema>& schema, arrow::MemoryPool* pool,
+                                         std::shared_ptr<arrow::RecordBatch>* result) {
+    if (chunk->num_columns() != schema->num_fields()) {
+        return Status::InvalidArgument("number fields not match");
+    }
+
+    std::vector<std::shared_ptr<arrow::Array>> arrays(slot_types.size());
+
+    for (auto i = 0; i < slot_types.size(); ++i) {
+        auto column = chunk->get_column_by_slot_id(slot_ids[i]);
+        if (column->only_null() || column->is_constant()) {
```

Comment:
check `column->is_constant()` is enough.

Reply:
ok